packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -77,25 +104,175 @@ function WorklistDataGrid(props) {
+  );
 
-  const gridRef = useCallback((node) => {
-    grid.current = node;
-    const focusedCell = grid.current.rows[focusedRow.current].cells[focusedCol.current];
+  const isCellSelected = (rowId, columnId) => (currentSelectedCell && currentSelectedCell.rowId === rowId && currentSelectedCell.columnId === columnId);
+
+  const setFocus = (rowIndex, colIndex, makeActiveElement) => {
+    let focusedCell = grid.current.rows[rowIndex].cells[colIndex];
+    if (isRowSelectionCell(colIndex) && focusedCell.firstChild) {
+      focusedCell = focusedCell.firstChild;
+    }
     focusedCell.tabIndex = 0;
-  }, []);
+    if (makeActiveElement && focusedCell.focus) {
+      focusedCell.focus();
+    }
+  };
 
-  const handleKeyDown = (event) => {
-    const currentFocusedRow = event.target.parentElement.rowIndex;
-    const currentFocusedCol = event.target.cellIndex;
+  const removeTabStop = (rowIndex, colIndex) => {
+    const cell = grid.current.rows[rowIndex].cells[colIndex];
+    // Remove Tab stop from previous cell in table that has focus and set it to the
+    // cell that was clicked.
+    cell.tabIndex = -1;
+    if (isRowSelectionCell(colIndex) && cell.firstChild) {
+      // Also remove tabstop from the checkbox in the case of a row selection cell.
+      cell.firstChild.tabIndex = -1;
+    }
+  };
+
+  const setFocusedRowCol = (newRowIndex, newColIndex, makeActiveElement) => {
+    removeTabStop(focusedRow.current, focusedCol.current);
+    focusedRow.current = newRowIndex;
+    focusedCol.current = newColIndex;
+    setFocus(newRowIndex, newColIndex, makeActiveElement);
+  };
+
+  useEffect(() => {
+    ariaLiveMsg.current = intl.formatMessage({ id: hasSelectableRows ? 'Terra.worklist-data-grid.row-selection-mode-enabled' : 'Terra.worklist-data-grid.row-selection-mode-disabled' });
+    // When row selection mode is turned on or off a row selection column is added or removed.
+    // Therefore, shift the focused cell to the left or right.
+    let newFocusCell = { row: focusedRow.current, col: (focusedCol.current + (hasSelectableRows ? 1 : -1)) };
+
+    if (hasSelectableRows && focusedRow.current === 0 && focusedCol.current === 0) {
+      // When row selection is turned on for the first time, default to the first row selection cell.
+      newFocusCell = { row: 1, col: 0 };
+    } else if (!hasSelectableRows && focusedCol.current === 0) {
+      // When row selection is turned off, if a cell in the row selection had focus, then
+      // refocus on the first cell in that row.
+      newFocusCell = { row: focusedRow.current, col: 0 };
+    }
+    setFocusedRowCol(newFocusCell.row, newFocusCell.col, false);
+    if (currentSelectedCell != null) {
+      setCurrentSelectedCell(null);
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [hasSelectableRows]);
+
+  const areRowsSelected = () => (
```

Comment:
This method would technically be, "isRowSelected", because you are searching for one match.

Reply:
I changed it to isAnyRowSelected.